apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -26,12 +27,50 @@ defmodule Aecore.Structures.SignedTx do
 
   @spec is_coinbase?(SignedTx.t()) :: boolean()
   def is_coinbase?(tx) do
-    tx.data.from_acc == nil && tx.signature == nil
+    if(match?(%SpendTx{}, tx.data)) do
+      tx.data.from_acc == nil && tx.signature == nil
+    else
+      false
+    end
   end
 
   @spec is_valid?(SignedTx.t()) :: boolean()
   def is_valid?(tx) do
-    tx.data.value >= 0 && tx.data.fee >= 0 && Keys.verify_tx(tx)
+    if(match?(%SpendTx{}, tx.data)) do
+      not_negative = tx.data.value >= 0
+      signature_valid = Keys.verify_tx(tx)
+      not_negative && signature_valid
+    else
+      true
```

Comment:
We have to validate signatures of Orace txs also. _This connects with my comments in Chain.ChainState_

Reply:
good catch